tests/system/feature_store/test_feature_store.py

```diff
@@ -2489,6 +2488,61 @@ def test_get_offline_for_two_feature_set_with_same_column_name(self):
         for key in res.to_dataframe().to_dict().keys():
             assert key in expected
 
+    @pytest.mark.parametrize(
+        "time_for_source, is_through_init, should_succeed, time_delta",
+        [
+            (None, None, True, None),
+            (datetime(2021, 5, 25, 10, 30, 29, 592000), False, True, None),
+            (datetime(2021, 5, 25, 10, 30, 29, 592000), True, True, None),
+            (datetime(2021, 5, 25, 10, 30, 29, 592000, timezone.utc), False, True, 0),
+            (datetime(2021, 5, 25, 10, 30, 29, 592000, timezone.utc), True, True, 0),
+            ("2021-05-25T10:30:29.abc", False, False, None),
+            ("2021-05-25T10:30:29.abc", True, False, None),
+            ("2021-05-25T10:30:29.592", False, True, None),
+            ("2021-05-25T10:30:29.592", True, True, None),
+            ("2021-05-25T10:30:29.592Z", False, True, 0),
+            ("2021-05-25T10:30:29.592Z", True, True, 0),
+            ("2021-05-25T10:30:29.592+00:00", False, True, 0),
+            ("2021-05-25T10:30:29.592+00:00", True, True, 0),
+            ("2021-05-25T10:30:29.592+02:00", False, True, timedelta(hours=2)),
+            ("2021-05-25T10:30:29.592+02:00", True, True, timedelta(hours=2)),
+        ],
+    )
+    def test_parquet_source_with_iso_start_or_end_time(
+        self, time_for_source, is_through_init, should_succeed, time_delta
+    ):
+        def actual_test(time_for_source, is_through_init, time_delta):
```

Comment:
If there's no more specific name then just the name of the test function simply go with (prefix underscore) `_test_parquet_source_with_iso_start_or_end_time`